src/db/results.rs

```diff
@@ -21,7 +21,21 @@ pub type DeleteBsos = SyncTimestamp;
 pub type DeleteBso = SyncTimestamp;
 pub type PutBso = SyncTimestamp;
 
-pub type CreateBatch = String;
+#[derive(Debug, Clone)]
+pub struct CreateBatch {
+    pub id: String,
+    pub size: Option<usize>,
```

Comment:
nit: it seems like size isn't needed in this struct. Unless I'm mistaken it seems to only be shared between `append`/`create` and `do_append`, I think those methods could just pass the value along to `do_append`

Reply:
I suppose. Part of my decision to include it here was to not keep adding arguments.